## there are special characters etc. that causes some bugs
## aim -> refactor & polish anime names

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
df = pd.read_csv("/content/anime_recommender_app_data.csv", index_col = 0)
df.head()

AnimeName  ClusterType
0  &quot;Bungaku Shoujo&quot; Kyou no Oyatsu: Hat...            3
1                 &quot;Bungaku Shoujo&quot; Memoire            0
2                   &quot;Bungaku Shoujo&quot; Movie            0
3                                   &quot;Eiji&quot;            3
4                               .hack//G.U. Returner            1

### disabled anime names for my model

In [49]:
# some special/unicode chars have in common = "&", ";" --> &quot; &#039; &amp;
disabled = ["&", "♥", "★", "♪", ";"]

of = ["&quot;", "&#039;", "&amp;", "♥", "★", "♪"]

to_change = list()

for i in df.AnimeName:
  # not efficient baby
  for j in disabled:
    if j in i:
      to_change.append(i)

to_change = set(to_change)
for i in to_change:
  print(i)

Nanatsu-iro★Drops
Gintama&#039;
Doraemon: Nobita&#039;s the Night Before a Wedding
Mobile Suit Gundam: Char&#039;s Counterattack
Kujibiki♥Unbalance
Tokyo Ghoul: &quot;Jack&quot;
Pokemon XY: New Year&#039;s Eve 2014 Super Mega Special
VS Knight Lamune &amp; 40 Fresh
Mahou Shoujo Madoka★Magica Movie 1: Hajimari no Monogatari
Dorami &amp; Doraemons: Space Land&#039;s Critical Event
Shakugan no Shana: Shana &amp; Yuuji no Naze Nani Shana! Nandemo Shitsumon-bako!
I&#039;&#039;s Pure Bonus
Black★Rock Shooter
Cat&#039;s Eye
The Tales of Rien Village #1: &quot;Toilet ni Ittokiya&quot;
Please ♥♥♥♥ Me!
Tengen Toppa Gurren Lagann: Kirameki★Yoko Box - Pieces of Sweet Stars
Ningen Shikkaku: Director&#039;s Cut-ban
Kiniro no Corda: Blue♪Sky
Working&#039;!! Announcement Specials
Mermaid&#039;s Scar
Prince of Tennis: Atobe&#039;s Gift
Yamato Nadeshiko Shichihenge♥
Black★Rock Shooter (TV)
Anne Happy♪
Love♥Love?
ChäoS;HEAd
Chains &amp; Rings
Marie &amp; Gali Episode Zero
Lance N&#039; Masques
Tokyo Ghou

In [45]:
len(set(to_change))

260

### i guess i am so dumb to fix it automatically yet ambitious to fix it, so i will manually rewrite 260 names :) let me know, if there is more efficient way

In [59]:
correct_dict = {
    "Nanatsu-iro★Drops" : "Nanatsu-iro Drops",
    "Gintama&#039;" : "Gintama",
    "Doraemon: Nobita&#039;s the Night Before a Wedding" : "Doraemon: Nobita\'s the Night Before a Wedding",
    "Mobile Suit Gundam: Char&#039;s Counterattack" : "Mobile Suit Gundam: Char's Counterattack",
    "Kujibiki♥Unbalance" : "Kujibiki Unbalance",
    "Tokyo Ghoul: &quot;Jack&quot;" : "Tokyo Ghoul: Jack",
    "Pokemon XY: New Year&#039;s Eve 2014 Super Mega Special" : "Pokemon XY: New Year's Eve 2014 Super Mega Special",
    "VS Knight Lamune &amp; 40 Fresh" : "VS Knight Lamune & 40 Fresh",
    "Mahou Shoujo Madoka★Magica Movie 1: Hajimari no Monogatari" : "Mahou Shoujo Madoka Magica Movie 1: Hajimari no Monogatari",
    "Dorami &amp; Doraemons: Space Land&#039;s Critical Event" : "Dorami & Doraemons: Space Land's Critical Event",
    "Shakugan no Shana: Shana &amp; Yuuji no Naze Nani Shana! Nandemo Shitsumon-bako!" : "Shakugan no Shana: Shana & Yuuji no Naze Nani Shana! Nandemo Shitsumon-bako!",
    "I&#039;&#039;s Pure Bonus" : "I''s Pure Bonus",
    "Black★Rock Shooter" : "Black Rock Shooter",
    "Cat&#039;s Eye" : "Cat's Eye",
    "The Tales of Rien Village #1: &quot;Toilet ni Ittokiya&quot;" : "The Tales of Rien Village #1: Toilet ni Ittokiya",
    "Please ♥♥♥♥ Me!" : "Please Me!",
    "Tengen Toppa Gurren Lagann: Kirameki★Yoko Box - Pieces of Sweet Stars" : "Tengen Toppa Gurren Lagann: Kirameki Yoko Box - Pieces of Sweet Stars",
    "Ningen Shikkaku: Director&#039;s Cut-ban" : "Ningen Shikkaku: Director's Cut-ban",
    "Kiniro no Corda: Blue♪Sky" : "Kiniro no Corda: Blue Sky",
    "Working&#039;!! Announcement Specials" : "Working!! Announcement Specials",
    "Mermaid&#039;s Scar" : "Mermaid's Scar",
    "Prince of Tennis: Atobe&#039;s Gift" : "Prince of Tennis: Atobe's Gift",
    "Yamato Nadeshiko Shichihenge♥" : "Yamato Nadeshiko Shichihenge",
    "Black★Rock Shooter (TV)" : "Black Rock Shooter (TV)",
    "Anne Happy♪" : "Anne Happy",
    "Love♥Love?" : "Love Love?",
    "ChäoS;HEAd" : "ChaoS HEAd",
    "Chains &amp; Rings" : "Chains & Rings",
    "Marie &amp; Gali Episode Zero" : "Marie & Gali Episode Zero",
    "Lance N&#039; Masques" : "Lance N' Masques",
    "Tokyo Ghoul: &quot;Pinto&quot;" : "Tokyo Ghoul: Pinto",
    "Chi&#039;s Sweet Home" : "Chi's Sweet Home",
    "Gintama&#039;: Enchousen" : "Gintama: Enchousen",
    "Gosenzo San&#039;e" : "Gosenzo San'e",
    "Hitozuma♪Kasumi-san" : "Hitozuma Kasumi-san",
    "Uta no☆Prince-sama♪ Maji Love Revolutions" : "Uta no Prince-sama Maji Love Revolutions",
    "Pokemon: Ash&#039;s Journey" : "Pokemon: Ash's Journey",
    "Dennou Sentai Voogie&#039;s★Angel" : "Dennou Sentai Voogie's Angel",
    "Panty &amp; Stocking in Sanitarybox" : "Panty & Stocking in Sanitarybox",
    "Chi&#039;s Sweet Home OVA" : "Chi's Sweet Home OVA",
    "Dog Days&#039; Specials" : "Dog Days Specials",
    "Sora no Method: Aru Shoujo no Kyuujitsu★" : "Sora no Method: Aru Shoujo no Kyuujitsu",
    "Shakotan★Boogie" : "Shakotan Boogie",
    "Seikimatsu★Darling" : "Seikimatsu Darling",
    "Uchuu Kyoudai: Apo&#039;s Dream" : "Uchuu Kyoudai: Apo's Dream",
    "Kitty&#039;s Paradise" : "Kitty's Paradise",
    "Tomorrow&#039;s Song" : "Tomorrow's Song",
    "Precure All Stars Movie: Haru no Carnival♪" : "Precure All Stars Movie: Haru no Carnival",
    "Onegai♪My Melody Kirara★" : "Onegai My Melody Kirara",
    "I&#039;&#039;s" : "I''s",
    "Lovely★Complex" : "Lovely Complex",
    "Queen&#039;s Blade: Rebellion Specials" : "Queen's Blade: Rebellion Specials",
    "Black★Rock Shooter (OVA)" : "Black Rock Shooter (OVA)",
    "Baka to Test to Shoukanjuu: Matsuri - Sentaku ni Yotte Tenkai ga Kawaru &quot;LIPS Eizou&quot;" : "Baka to Test to Shoukanjuu: Matsuri - Sentaku ni Yotte Tenkai ga Kawaru LIPS Eizou",
    "B&#039;T X Neo" : "B'T X Neo",
    "Queen&#039;s Blade: Rurou no Senshi" : "Queen's Blade: Rurou no Senshi",
    "Kyoukai no Kanata Movie: I&#039;ll Be Here - Mirai-hen" : "Kyoukai no Kanata Movie: I'll Be Here - Mirai-hen",
    "Sore Ike! Anpanman: Franken-Robo-kun&#039;s Surprised Christmas" : "Sore Ike! Anpanman: Franken-Robo-kun's Surprised Christmas",
    "Detective Conan: Conan vs. Kid - Shark &amp; Jewel" : "Detective Conan: Conan vs. Kid - Shark & Jewel",
    "Atashin&#039;chi" : "Atashin'chi",
    "Boku no Imouto wa &quot;Osaka Okan&quot;: Haishin Gentei Osaka Okan." : "Boku no Imouto wa Osaka Okan: Haishin Gentei Osaka Okan",
    "Joshikousei: Girl&#039;s High" : "Joshikousei: Girl's High",
    "Suite Precure♪ Movie: Torimodose! Kokoro ga Tsunaku Kiseki no Melody♪" : "Suite Precure Movie: Torimodose! Kokoro ga Tsunaku Kiseki no Melody",
    "Penguin Musume♥Heart Special" : "Penguin Musume Heart Special",
    "Precure All Stars Movie: Minna de Utau♪ - Kiseki no Mahou" : "Precure All Stars Movie: Minna de Utau - Kiseki no Mahou",
    "Tales of Symphonia The Animation: Tethe&#039;alla-hen Specials" : "Tales of Symphonia The Animation: Tethe'alla-hen Specials",
    "Mad★Bull 34" : "Mad Bull 34",
    "Queen&#039;s Blade OVA Specials" : "Queen's Blade OVA Specials",
    "Bakusou Kyoudai Let&#039;s &amp; Go WGP" : "Bakusou Kyoudai Let's & Go WGP",
    "Love Hina: Motoko no Sentaku, Koi ka Ken... &quot;Naku na&quot;" : "Love Hina: Motoko no Sentaku, Koi ka Ken... Naku na",
    "NG Knight Ramune &amp; 40" : "NG Knight Ramune & 40",
    "Tiger &amp; Bunny" : "Tiger & Bunny",
    "Aquarian Age: Saga II - Don&#039;t Forget Me..." : "Aquarian Age: Saga II - Don't Forget Me...",
    "Viper&#039;s Creed" : "Viper's Creed",
    "Tokumu Sousakan Rei &amp; Fuko" : "Tokumu Sousakan Rei & Fuko",
    "Queen&#039;s Blade: Rebellion" : "Queen's Blade: Rebellion",
    "Persona 3 the Movie 2: Midsummer Knight&#039;s Dream" : "Persona 3 the Movie 2: Midsummer Knight's Dream",
    "Doraemon&#039;s Time Capsule for 2001" : "Doraemon's Time Capsule for 2001",
    "Top wo Nerae! &amp; Top wo Nerae 2! Gattai Movie!!" : "Top wo Nerae! & Top wo Nerae 2! Gattai Movie!!",
    "Ketsuekigata-kun! Featuring Yuri &amp; Nasuno" : "Ketsuekigata-kun! Featuring Yuri & Nasuno",
    "Let&#039;s Nupu Nupu" : "Let's Nupu Nupu",
    "Asura Cryin&#039;" : "Asura Cryin'",
    "Visions of Frank: Short Films by Japan&#039;s Most Audacious Animators" : "Visions of Frank: Short Films by Japan's Most Audacious Animators",
    "Shin Atashin&#039;chi" : "Shin Atashin'chi",
    "Mahou Shoujo Lyrical Nanoha A&#039;s" : "Mahou Shoujo Lyrical Nanoha A's",
    "Sketchbook: Full Color&#039;s Picture Drama" : "Sketchbook: Full Color's Picture Drama",
    "Mahou Shoujo Lyrical Nanoha: The Movie 2nd A&#039;s" : "Mahou Shoujo Lyrical Nanoha: The Movie 2nd A's",
    "Bakusou Kyoudai Let&#039;s &amp; Go MAX" : "Bakusou Kyoudai Let's & Go MAX",
    "Le Chevalier D&#039;Eon" : "Le Chevalier D'Eon",
    "Dog Days&#039;&#039;" : "Dog Days''",
    "Queen&#039;s Blade: Rurou no Senshi Specials" : "Queen's Blade: Rurou no Senshi Specials",
    "Sonic★the★Hedgehog" : "Sonic the Hedgehog",
    "Gin&#039;iro no Olynsis: Tokito" : "Gin'iro no Olynsis: Tokito",
    "Gyakuten Saiban: Sono &quot;Shinjitsu&quot;, Igi Ari!" : "Gyakuten Saiban: Sono Shinjitsu, Igi Ari!",
    "It&#039;s a Rumic World: 50th Anniversary Weekly★Shonen Sunday" : "It's a Rumic World: 50th Anniversary Weekly Shonen Sunday",
    "Mobile Suit Gundam: The 08th MS Team - Miller&#039;s Report" : "Mobile Suit Gundam: The 08th MS Team - Miller's Report",
    "Penguin Musume♥Heart" : "Penguin Musume Heart",
    "009-1: R&amp;B" : "009-1: R&B",
    "Black Lagoon: Roberta&#039;s Blood Trail" : "Black Lagoon: Roberta's Blood Trail",
    "Uta no☆Prince-sama♪ Maji Love 2000%" : "Uta no Prince-sama Maji Love 2000%",
    "Rin x Sen + Ran -&gt; Sem: Cross Mix" : "Rin x Sen + Ran -> Sem: Cross Mix",
    "Marie &amp; Gali" : "Marie & Gali",
    "&quot;Eiji&quot;" : "Eiji",
    "Pokemon Omega Ruby &amp; Alpha Sapphire: Mega Special Animation" : "Pokemon Omega Ruby & Alpha Sapphire: Mega Special Animation",
    "Honoo no Haramase Paidol My★Star Gakuen Z The Animation" : "Honoo no Haramase Paidol My Star Gakuen Z The Animation",
    "Detective Conan OVA 06: Follow the Vanished Diamond! Conan &amp; Heiji vs. Kid!" : "Detective Conan OVA 06: Follow the Vanished Diamond! Conan &a Heiji vs. Kid!",
    "Queen&#039;s Blade: Grimoire" : "Queen's Blade: Grimoire",
    "Ore ga Ojousama Gakkou ni &quot;Shomin Sample&quot; Toshite Gets♥Sareta Ken Specials" : "Ore ga Ojousama Gakkou ni Shomin Sample Toshite Gets Sareta Ken Specials",
    "Glass no Kantai: La Legende du Vent de l&#039;Univers" : "Glass no Kantai: La Legende du Vent de l'Univers",
    "Ginga Ojousama Densetsu Yuna: Shin&#039;en no Fairy" : "Ginga Ojousama Densetsu Yuna: Shin'en no Fairy",
    "Yu☆Gi☆Oh! 5D&#039;s" : "Yu Gi Oh! 5D's",
    "Tiger &amp; Bunny Movie 1: The Beginning" : "Tiger & Bunny Movie 1: The Beginning",
    "Yoshinaga-san&#039;chi no Gargoyle" : "Yoshinaga-san'chi no Gargoyle",
    "Steins;Gate Movie: Fuka Ryouiki no Déjà vu" : "Steins;Gate Movie: Fuka Ryouiki no Deja vu",
    "Escha &amp; Logy no Atelier: Tasogare no Sora no Renkinjutsushi" : "Escha & Logy no Atelier: Tasogare no Sora no Renkinjutsushi",
    "Love♥Love? Specials" : "Love Love? Specials",
    "Luck &amp; Logic" : "Luck & Logic",
    "D.C. I&amp;II P.S.P. Re-Animated" : "D.C. I&II P.S.P. Re-Animated",
    "Tiger &amp; Bunny Pilot" : "Tiger & Bunny Pilot",
    "Chi&#039;s Sweet Home: Atarashii Ouchi" : "Chi's Sweet Home: Atarashii Ouchi",
    "Mahou Shoujo Madoka★Magica Movie 3: Hangyaku no Monogatari" : "Mahou Shoujo Madoka Magica Movie 3: Hangyaku no Monogatari",
    "Ero Manga Mitai na Koi Shiyo: Let&#039;s Fall in Love The Ero-manga" : "Ero Manga Mitai na Koi Shiyo: Let's Fall in Love The Ero-manga",
    "Mahou Shoujo Madoka★Magica" : "Mahou Shoujo Madoka Magica",
    "&quot;Bungaku Shoujo&quot; Kyou no Oyatsu: Hatsukoi" : "Bungaku Shoujo Kyou no Oyatsu: Hatsukoi",
    "Recorder to Randoseru Re♪" : "Recorder to Randoseru Re",
    "Sketchbook: Full Color&#039;s" : "Sketchbook: Full Color's",
    "Lupin III: Episode 0 &quot;First Contact&quot;" : "Lupin 3: Episode 0 First Contact",
    "Pokemon XY&amp;Z: Subete no Nazo wo Tokiakase!" : "Pokemon XY&Z: Subete no Nazo wo Tokiakase!",
    "Suite Precure♪" : "Suite Precure", 
    "City Hunter &#039;91" : "City Hunter '91",
    "I&#039;&#039;s Pure" : "I''s Pure",
    "Joshikousei: Girl&#039;s High Specials" : "Joshikousei: Girl's High Specials",
    "Sailor Senshi Venus♥Five" : "Sailor Senshi Venus Five",
    "Nikutai Ten&#039;i" : "Nikutai Ten'i",
    "Love Live! School Idol Project: μ&#039;s →NEXT LoveLive! 2014 - Endless Parade Makuai Drama" : "Love Live! School Idol Project: Muse's NEXT LoveLive! 2014 - Endless Parade Makuai Drama",
    "Hikyou Tanken Fam &amp; Ihrlie" : "Hikyou Tanken Fam & Ihrlie",
    "Boku no Imouto wa &quot;Osaka Okan&quot;: Uchi no Oniichan wa Tokyo Rule" : "Boku no Imouto wa Osaka Okan: Uchi no Oniichan wa Tokyo Rule",
    "Seme♥Chichi" : "Seme Chichi",
    "Sister Princess: Re Pure Character&#039;s" : "Sister Princess: Re Pure Character's",
    "Tiger &amp; Bunny Movie 2: The Rising" : "Tiger & Bunny Movie 2: The Rising",
    "Pokemon XY&amp;Z" : "Pokemon XY&Z",
    "Detective Conan OVA 08: High School Girl Detective Sonoko Suzuki&#039;s Case Files" : "Detective Conan OVA 08: High School Girl Detective Sonoko Suzuki's Case Files",
    "Lupin III: Ai no Da Capo - Fujiko&#039;s Unlucky Days" : "Lupin 3: Ai no Da Capo - Fujiko's Unlucky Days",
    "Dae Jang Geum: Jang Geum&#039;s Dream" : "Dae Jang Geum: Jang Geum's Dream",
    "Queen&#039;s Blade OVA" : "Queen's Blade OVA",
    "Precure All Stars Movie: Haru no Carnival♪ - Ima Koko kara" : "Precure All Stars Movie: Haru no Carnival - Ima Koko kara",
    "Precure All Stars DX the Dance Live♥: Miracle Dance Stage e Youkoso" : "Precure All Stars DX the Dance Live: Miracle Dance Stage e Youkoso",
    "Ore ga Ojousama Gakkou ni &quot;Shomin Sample&quot; Toshite Gets♥Sareta Ken" : "Ore ga Ojousama Gakkou ni Shomin Sample Toshite Gets Sareta Ken",
    "Kaichou wa Maid-sama!: Goshujinsama to Asonjao♥" : "Kaichou wa Maid-sama!: Goshujinsama to Asonjao",
    "Persona 4 the Animation: A Brief Lesson on Izanagi &amp; Izanami" : "Persona 4 the Animation: A Brief Lesson on Izanagi & Izanami",
    "VS Knight Lamune &amp; 40 Fire" : "VS Knight Lamune & 40 Fire",
    "E&#039;s Otherwise" : "E's Otherwise",
    "Suzy&#039;s Zoo: Daisuki! Witzy - Happy Birthday" : "Suzy's Zoo: Daisuki! Witzy - Happy Birthday",
    "Je T&#039;aime" : "Je T'aime",
    "Working&#039;!!" : "Working!!",
    "Mahou Shoujo Madoka★Magica Movie 3: Hangyaku no Monogatari - Magica Quartet x Nisioisin" : "Mahou Shoujo Madoka Magica Movie 3: Hangyaku no Monogatari - Magica Quartet x Nisioisin",
    "Issho ni Training Ofuro: Bathtime with Hinako &amp; Hiyoko" : "Issho ni Training Ofuro: Bathtime with Hinako & Hiyoko",
    "Pokemon Diamond &amp; Pearl Specials" : "Pokemon Diamond & Pearl Specials",
    "Hayate no Gotoku! Can&#039;t Take My Eyes Off You" : "Hayate no Gotoku! Can't Take My Eyes Off You",
    "Blade &amp; Soul Specials" : "Blade & Soul Specials",
    "Violence Jack: Hell&#039;s Wind-hen" : "Violence Jack: Hell's Wind-hen",
    "Pokemon Diamond &amp; Pearl: Giratina to Sora no Hanataba Sheimi" : "Pokemon Diamond & Pearl: Giratina to Sora no Hanataba Sheimi",
    "Pokemon Diamond &amp; Pearl" : "Pokemon Diamond & Pearl",
    "Tekkamen wo Oe: &quot;d&#039;Artagnan Monogatari&quot; yori" : "Tekkamen wo Oe: d'Artagnan Monogatari yori",
    "Queen&#039;s Blade: Gyokuza wo Tsugu Mono Specials" : "Queen's Blade: Gyokuza wo Tsugu Mono Specials",
    "Eden&#039;s Bowy" : "Eden's Bowy",
    "Tales of Symphonia The Animation: Tethe&#039;alla-hen" : "Tales of Symphonia The Animation: Tethe'alla-hen",
    "Puzzle &amp; Dragons CM" : "Puzzle & Dragons CM",
    "Itadaki! Seieki♥ Special" : "Itadaki! Seieki Special",
    "Ikoku Meiro no Croisée: Yune &amp; Alice" : "Ikoku Meiro no Croisee: Yune & Alice",
    "Queen&#039;s Blade Rebellion vs. Hagure Yuusha no Aesthetica" : "Queen's Blade Rebellion vs. Hagure Yuusha no Aesthetica",
    "Mahou Shoujo Sonico★Magica" : "Mahou Shoujo Sonico Magica",
    "Mahoromatic: I&#039;m Home!" : "Mahoromatic: I'm Home!",
    "Dogs: Bullets &amp; Carnage" : "Dogs: Bullets & Carnage",
    "Eat-Man &#039;98" : "Eat-Man & '98",
    "Itadaki! Seieki♥" : "Itadaki! Seieki",
    "Uta no☆Prince-sama♪ Maji Love 2000%: Shining Star Xmas" : "Uta no Prince-sama Maji Love 2000%: Shining Star Xmas",
    "Onegai My Melody Sukkiri♪" : "Onegai My Melody Sukkiri",
    "Wanna-Be&#039;s" : "Wanna-Be's",
    "Fukubiki! Triangle: Futaba wa Atafuta ♥" : "Fukubiki! Triangle: Futaba wa Atafuta",
    "Dragon Nest: Warriors&#039; Dawn" : "Dragon Nest: Warriors' Dawn",
    "Dr. Slump Movie 02: &quot;Hoyoyo!&quot; Uchuu Daibouken" : "Dr. Slump Movie 02: Hoyoyo! Uchuu Daibouken",
    "I&#039;m Here with You" : "I'm Here with You",
    "&quot;Bungaku Shoujo&quot; Memoire" : "Bungaku Shoujo Memoire",
    "Mahou Shoujo Madoka★Magica Movie 2: Eien no Monogatari" : "Mahou Shoujo Madoka Magica Movie 2: Eien no Monogatari",
    "Ninja &amp; Soldier" : "Ninja & Soldier",
    "Dog Days&#039;" : "Dog Days'",
    "Happy☆Lesson: Hoka Hoka★Kanna to Futari Kiri" : "Happy Lesson: Hoka Hoka Kanna to Futari Kiri",
    "Recorder to Randoseru Do♪" : "Recorder to Randoseru Do",
    "Yu☆Gi☆Oh! 5D&#039;s: Shinkasuru Kettou! Stardust vs. Red Demon&#039;s" : "Yu Gi Oh! 5D's: Shinkasuru Kettou! Stardust vs. Red Demon's",
    "Kid&#039;s Castle" : "Kid's Castle",
    "B&#039;T X" : "B'T X",
    "Neon Genesis Evangelion: Death &amp; Rebirth" : "Neon Genesis Evangelion: Death & Rebirth",
    "Arata naru Sekai: World&#039;s/Start/Load/End" : "Arata naru Sekai: World's/Start/Load/End",
    "Pokemon Diamond &amp; Pearl: Dialga vs. Palkia vs. Darkrai" : "Pokemon Diamond & Pearl: Dialga vs. Palkia vs. Darkrai",
    "The iDOLM@STER Million Live! &quot;Dreaming!&quot; Animation PV" : "The iDOLM@STER Million Live! Dreaming! Animation PV",
    "Naruto Shippuuden: Shippuu! &quot;Konoha Gakuen&quot; Den" : "Naruto Shippuuden: Shippuu! Konoha Gakuen Den",
    "Hal &amp; Bons" : "Hal & Bons",
    "Queen&#039;s Blade: Gyokuza wo Tsugu Mono" : "Queen's Blade: Gyokuza wo Tsugu Mono",
    "Bakusou Kyoudai Let&#039;s &amp; Go" : "Bakusou Kyoudai Let's & Go",
    "Blade &amp; Soul" : "Blade & Soul",
    "6♥Princess" : "6 Princess",
    "Luigi&#039;s Toy Adventure" : "Luigi's Toy Adventure",
    "Panty &amp; Stocking with Garterbelt" : "Panty & Stocking with Garterbelt",
    "Ail Maniax: Inma Seifukugari &amp; Majogari no Yoru ni" : "Ail Maniax: Inma Seifukugari & Majogari no Yoru ni",
    "Pokemon Diamond &amp; Pearl: Genei no Hasha Zoroark" : "Pokemon Diamond & Pearl: Genei no Hasha Zoroark",
    "Pokemon Diamond &amp; Pearl: Arceus Choukoku no Jikuu e" : "Pokemon Diamond & Pearl: Arceus Choukoku no Jikuu e",
    "Moegaku★5" : "Moegaku 5",
    "Boku no Imouto wa &quot;Osaka Okan&quot;" : "Boku no Imouto wa Osaka Okan",
    "Yuru Yuri♪♪" : "Yuru Yuri",
    "Aishiteruze Baby★★" : "Aishiteruze Baby",
    "Heppoko Jikken Animation Excel♥Saga" : "Heppoko Jikken Animation Excel Saga",
    "Marriage Blue: &quot;Konyakusha ga Iru no ni, Doushite Konna Otoko ni......&quot;" : "Marriage Blue: Konyakusha ga Iru no ni, Doushite Konna Otoko ni",
    "H. P. Lovecraft&#039;s The Dunwich Horror and Other Stories" : "H. P. Lovecraft's The Dunwich Horror and Other Stories",
    "Be-Boy Kidnapp&#039;n Idol" : "Be-Boy Kidnapp'n Idol",
    "Dragon&#039;s Heaven" : "Dragon's Heaven",
    "Marie &amp; Gali Special" : "Marie & Gali Special",
    "Uta no☆Prince-sama♪ Maji Love 1000%" : "Uta no Prince-sama Maji Love 1000%",
    "Asura Cryin&#039; 2" : "Asura Cryin' 2",
    "Dr. Slump Movie 11: Dr. Mashirito &amp; Abale-chan" : "Dr. Slump Movie 11: Dr. Mashirito & Abale-chan",
    "Mahou Shoujo Madoka★Magica: Concept Movie" : "Mahou Shoujo Madoka Magica: Concept Movie",
    "Ran-&gt;Sem: Hakudaku Delmo Tsuma no Miira Tori" : "Ran->Sem: Hakudaku Delmo Tsuma no Miira Tori",
    "Marie &amp; Gali ver. 2.0" : "Marie & Gali ver. 2.0",
    "Oyome-sama Honey Days♥" : "Oyome-sama Honey Days",
    "Sparrow&#039;s Hotel" : "Sparrow's Hotel",
    "Kyoukai no Kanata Movie: I&#039;ll Be Here - Kako-hen - Yakusoku no Kizuna" : "Kyoukai no Kanata Movie: I'll Be Here - Kako-hen - Yakusoku no Kizuna",
    "Touch 3: Kimi ga Toorisugita Ato ni - Don&#039;t Pass Me By" : "Touch 3: Kimi ga Toorisugita Ato ni - Don't Pass Me By",
    "Suzy&#039;s Zoo: Daisuki! Witzy" : "Suzy's Zoo: Daisuki! Witzy",
    "eX-Driver: Nina &amp; Rei Danger Zone" : "eX-Driver: Nina & Rei Danger Zone",
    "Detective Conan OVA 05: The Target is Kogoro! The Detective Boys&#039; Secret Investigation" : "Detective Conan OVA 05: The Target is Kogoro! The Detective Boys' Secret Investigation",
    "3Ping Lovers!☆Ippu Nisai no Sekai e Youkoso♪ The Animation" : "3Ping Lovers! Ippu Nisai no Sekai e Youkoso The Animation",
    "Flag Director&#039;s Edition: Issenman no Kufura no Kiroku" : "Flag Director's Edition: Issenman no Kufura no Kiroku",
    "I&#039;ll/CKBC" : "I'll/CKBC",
    "Glass no Kantai: La Legende du Vent de l&#039;Univers Specials" : "Glass no Kantai: La Legende du Vent de l'Univers Specials",
    "Queen&#039;s Blade: Utsukushiki Toushitachi" : "Queen's Blade: Utsukushiki Toushitachi",
    "Pokemon XY&amp;Z Specials" : "Pokemon XY&Z Specials",
    "Yokoo&#039;s 3 Animation Films" : "Yokoo's 3 Animation Films",
    "Honoo no Alpenrose: Judy &amp; Randy" : "Honoo no Alpenrose: Judy & Randy",
    "Mobile Suit SD Gundam&#039;s Counterattack" : "Mobile Suit SD Gundam's Counterattack",
    "Wolf&#039;s Rain OVA" : "Wolf's Rain OVA",
    "Figure 17: Tsubasa &amp; Hikaru" : "Figure 17: Tsubasa & Hikaru",
    "PriPara Movie: Mi~nna no Akogare♪ Let&#039;s Go☆Prix Paris" : "PriPara Movie: Minna no Akogare Let's Go Prix Paris",
    "Wolf&#039;s Rain" : "Wolf's Rain",
    "Kyoukai no Kanata Movie: I&#039;ll Be Here - Kako-hen" : "Kyoukai no Kanata Movie: I'll Be Here - Kako-hen",
    "Shin Megami Tensei Devil Children: Light &amp; Dark" : "Shin Megami Tensei Devil Children: Light & Dark",
    "Love Live! School Idol Project: μ&#039;s →NEXT LoveLive! 2014 - Endless Parade Encore Animation" : "Love Live! School Idol Project: Muse's NEXT LoveLive! 2014 - Endless Parade Encore Animation",
    "Pokemon the Movie XY&amp;Z: Volcanion to Karakuri no Magiana" : "Pokemon the Movie XY&Z: Volcanion to Karakuri no Magiana",
    "Mou Hasamazu ni wa Irarenai♥: Hoshi ni Onegai shitara Konnani Okkiku Nacchatta!" : "Mou Hasamazu ni wa Irarenai: Hoshi ni Onegai shitara Konnani Okkiku Nacchatta!",
    "&quot;Bungaku Shoujo&quot; Movie" : "Bungaku Shoujo Movie",
    "Angel&#039;s Feather" : "Angel's Feather",
    "Tony&#039;s Heroine Series: Kanojo wa Hanayome Kouhosei? Cinderella Collection" : "Tony's Heroine Series: Kanojo wa Hanayome Kouhosei? Cinderella Collection",
    "Bonjour♪Koiaji Pâtisserie" : "Bonjour Koiaji Patisserie",
}

In [60]:
df.AnimeName.map(correct_dict)

0       Bungaku Shoujo Kyou no Oyatsu: Hatsukoi
1                        Bungaku Shoujo Memoire
2                          Bungaku Shoujo Movie
3                                          Eiji
4                                           NaN
                         ...                   
7847                                        NaN
7848                                        NaN
7849                                        NaN
7850                                        NaN
7851                                        NaN
Name: AnimeName, Length: 7852, dtype: object

In [145]:
df.replace(correct_dict, inplace=True)

In [146]:
df.head()

AnimeName  ClusterType
0  Bungaku Shoujo Kyou no Oyatsu: Hatsukoi            3
1                   Bungaku Shoujo Memoire            0
2                     Bungaku Shoujo Movie            0
3                                     Eiji            3
4                     .hack//G.U. Returner            1

In [149]:
# save last data
from google.colab import files

df.to_csv("/content/last_anime_recommender_app_data.csv")
files.download("/content/last_anime_recommender_app_data.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>